src/components/Profile/ProfileCardList/ProfileCardList.tsx

```diff
@@ -1,84 +1,156 @@
-"use client";
+'use client';
 
-import TitleText from "@/components/Common/TitleText/TitleText";
-import SquareBox from "@/components/Ui/SquareBox/SquareBox";
-import { cn, generateColor } from "@/utils";
-import { Profile } from "contentlayer/generated";
-import Image from "next/image";
-import Link from "next/link";
+import SpacingDivider from '@/components/Common/SpacingDivider/SpacingDivider';
+import TitleText from '@/components/Common/TitleText/TitleText';
+import SquareBox from '@/components/Ui/SquareBox/SquareBox';
+import { cn, generateColor } from '@/utils';
+import { Profile } from 'contentlayer/generated';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useSearchParams, useRouter } from 'next/navigation';
+import { useMemo } from 'react';
 
 type TPropsProfileCardList = {
   profiles: Profile[];
 };
 
+const Tag = ({ tag }: { tag: string }) => {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+
+  const tmpSearchParam = new URLSearchParams(searchParams.toString());
+  tmpSearchParam.set('tag', tag);
+
+  return (
+    <TitleText
+      className={cn(
+        'inline-block cursor-pointer text-[10px] px-2 py-1 rounded-full mb-1 mr-[4px] bg-opacity-50 hover:bg-opacity-80',
+        generateColor()
+      )}
+      key={tag}
+      tag="span"
+      onClick={(e) => {
+        e.preventDefault();
+        router.push(`/profile?${tmpSearchParam.toString()}`);
+      }}
+    >
+      {tag}
+    </TitleText>
+  );
+};
+
 const ProfileCardList = ({ profiles }: TPropsProfileCardList) => {
+  const searchParams = useSearchParams();
+  const tag = searchParams.get('tag') ?? '';
+
+  const uniqueTags = useMemo(() => {
+    const tags: string[] = [];
+    const tmpSet = new Set(profiles.map((profile) => profile.tags ?? []).flat());
+    tmpSet.forEach((tag) => void tags.push(tag));
+    return tags;
+  }, [profiles]);
+
+  const filteredProfiles = useMemo(() => {
```

Comment:
bro @lwinmoepaing 

tag filtering is a little bit hacky. Because I see some people write something like `vue`, `vuejs`, `vue.js`, etc.  So, this function attempt to check those variants.